dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/Resource.java

```diff
@@ -68,10 +68,15 @@
    */
   private int userId;
 
-  /**
-   * resource type
-   */
-  private ResourceType type;
+    /**
+     * user Name
+     */
+    private String userName;
```

Comment:
I suggest you can create a new class called `ResourceDTO`, separate from the entities in the database, the advantage is that it will be easy to extend in the future. 

This is just a suggestion.

Reply:
It is recommended to modify, there are too many such confusing object definitions that are not easy to maintain.

Reply:
Thanks @CalvinKirs 
+ I have tried to use a new class, named `ResourceWrapper`, to wrap the Resource and maintain a field like userName, but I found that this may lead to the code to be more difficult to maintain. As the owner must be updated in both `queryResourceListPaging` and `updateResource`. 

+ The `updateResource` method is very complicated (this method has nearly 200 lines)and includes the processing of `File Resource` and `UDF Resource`. 
+ If I use `ResourceWrapper` in this method, I also need to determine whether the currently processed resource is FileResource, which will make the code more difficult to maintain. 
So I gave up this idea, What do you think is more appropriate?

```java
    @Override
    public Map<String, Object> queryResourceListPaging(User loginUser, int directoryId, ResourceType type, String searchVal, Integer pageNo, Integer pageSize) {

        HashMap<String, Object> result = new HashMap<>();
        Page<Resource> page = new Page<>(pageNo, pageSize);
        int userId = loginUser.getId();
        if (isAdmin(loginUser)) {
            userId = 0;
        }
        if (directoryId != -1) {
            Resource directory = resourcesMapper.selectById(directoryId);
            if (directory == null) {
                putMsg(result, Status.RESOURCE_NOT_EXIST);
                return result;
            }
        }

        List<Integer> resourcesIds = resourceUserMapper.queryResourcesIdListByUserIdAndPerm(userId, 0);

        IPage<Resource> resourceIPage = resourcesMapper.queryResourcePaging(page, userId, directoryId, type.ordinal(), searchVal, resourcesIds);

        PageInfo<Resource> pageInfo = new PageInfo<>(pageNo, pageSize);
        pageInfo.setTotalCount((int) resourceIPage.getTotal());
        pageInfo.setLists(resourceIPage.getRecords());
        result.put(Constants.DATA_LIST, pageInfo);
        putMsg(result, Status.SUCCESS);
        return result;
    }


    @Override
    @Transactional(rollbackFor = Exception.class)
    public Result<Object> updateResource(User loginUser,
                                         int resourceId,
                                         String name,
                                         String desc,
                                         ResourceType type,
                                         MultipartFile file) {
        Result<Object> result = checkResourceUploadStartupState();
        if (result.isFailed()) {
            return result;
        }

        Resource resource = resourcesMapper.queryResourceAndOwnerByResourceId(resourceId);
       // 省略
    }

```

Reply:
@kyoty I think you should separate the datasource entity from the front-end return entity. If you represent it in one class, you might meet the below problems.
1. If you need to change the paremeters that the front-end needs(the current case), you need to modify this class. 
2. if the table in the database change, you also need to modify this class. 
3. If you want to remove some parameters returned to the front-end, you might also add @JsonIgnore in this class. 

Over time, the class will become more and more complex, it's difficult to know the actual definition in the database.
The recommendation is to use a separate entity(DO) to represent the table object in the database, and convert him into DTO  and return to the front end. DTO may have richer attributes.
But if you do this, you may not be able to use `join` in sql(`join` is not a good way).

Reply:
Thanks @ruanwenjun , @CalvinKirs .
I've updated  by add a new class ResourceDTO and update the data returned to the front-end by using ResourceDTO instead of Resource.
can you have a look again? This pr has been hang for a long time.